--- libinet6/configure.in.orig	Wed Dec  4 14:33:39 2002
+++ libinet6/configure.in	Fri Feb 28 11:40:37 2003
@@ -53,7 +53,7 @@
 AC_CHECK_HEADERS(netash/ash.h netatalk/at.h netax25/ax25.h neteconet/ec.h netipx/ipx.h netrose/rose.h)
 
 AC_MSG_CHECKING([for glibc version])
-AC_EGREP_CPP(%%%yes%%%, [dnl
+AC_EGREP_CPP(%%%yes%%%, [
 #include <features.h>
 #if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ == 1
 %%%yes%%%
@@ -62,7 +62,7 @@
 	glibc_version=2.1
 	usagi_libc=21
 ],[
-	AC_EGREP_CPP(%%%yes%%%, [dnl
+	AC_EGREP_CPP(%%%yes%%%, [
 #include <features.h>
 #if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ == 2
 %%%yes%%%
@@ -71,7 +71,7 @@
 		glibc_version=2.2
 		usagi_libc=22
 	],[
-		AC_EGREP_CPP(%%%yes%%%, [dnl
+		AC_EGREP_CPP(%%%yes%%%, [
 #include <features.h>
 #if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ == 3
 %%%yes%%%
