--- libinet6/Makefile.in.orig	Thu Jan  9 10:07:32 2003
+++ libinet6/Makefile.in	Fri Feb 28 11:40:06 2003
@@ -38,7 +38,7 @@
 PERL = @perl@
 LORDER = @lorder@
 CFLAGS = -D_LIBC_REENTRANT -D_IO_MTSAFE_IO -D_USAGI -D_USAGI_LIBINET6=$(usagi_libc) -D_GNU_SOURCE -DCOMPAT_RFC2292 -Iinclude \
-		-D__ss_family=ss_family -D__ss_len=ss_len
+		-D__ss_family=ss_family -D__ss_len=ss_len $(OPT)
 DEFS = @DEFS@
 
 usagi_libc=@usagi_libc@
